pandapipes/plotting/generic_geodata.py

```diff
@@ -53,19 +51,18 @@ def build_igraph_from_ppipes(net, junctions=None):
     pp_junction_mapping = dict(list(zip(junction_index, list(range(nr_junctions)))))
 
     mask = _get_element_mask_from_nodes(net, "pipe", ["from_junction", "to_junction"], junctions)
-    for junction in net.pipe[mask].itertuples():
-        g.add_edge(pp_junction_mapping[junction.from_junction],
-                   pp_junction_mapping[junction.to_junction],
-                   weight=junction.length_km)
+    for pipe in net.pipe[mask].itertuples():
+        g.add_edge(pp_junction_mapping[pipe.from_junction], pp_junction_mapping[pipe.to_junction],
+                   weight=pipe.length_km)
 
     for comp in net['component_list']:
-        if comp in [Source, Sink, ExtGrid, Pipe, Junction]:
+        if not isinstance(comp, (BranchComponent, CirculationPump)):
             continue
         mask = _get_element_mask_from_nodes(
-            net, comp.table_name(), ["from_junction", "to_junction"], junctions)
+            net, comp.table_name(), comp.from_to_node_cols(), junctions)
+        fjc, tjc = comp.from_to_node_cols()
```

Comment:
Put this first, then you do not need to call the function twice.

Reply:
Did that.